### ISSUE TITLE
[v6rc3] Submitting programatically

### ISSUE THREAD

**Issue by gschwa** [erikras/redux-form#1304]
In the documentation for alpha15 (cannot find the rc3 docs) it mentions that the `submit()` function will be passed as a prop to the component so we could submit programmatically. As of rc3 the function is not being passed.

see: http://redux-form.com/6.0.0-alpha.15/docs/api/ReduxForm.md/

What is the correct way to submit the form programmatically? (is it only using a `ref`?)

**Comment by erikras** [comment]
No, it's only ever been an instance method (on the component, see also https://github.com/erikras/redux-form/issues/1209#issuecomment-230016335). To submit programmatically from inside your form component, you could:

#### 1. Use the actual form DOM component.

```js
<form ref="form" onSubmit={handleSubmit}>
  ...
</form>
```
and
```js
this.refs.form.submit()
```

#### 2. Call handle submit directly.

```js
const submitter = this.props.handleSubmit(this.props.onSubmit)
submitter() // submits
```

The reason you cannot call `handleSubmit()` directly is that it is expecting either a submit function or an `event` to cancel as its first parameter.

**Comment by gschwa** [comment]
Makes perfect sense. Thank you.

**Comment by goodbomb** [comment]
@erikras Your first solution causes the entire page to refresh. It would seem that the only way to submit the form programmatically is to pass the "onSubmit" function from the parent, which is rather annoying IMO.

I'd like to be able to submit my form onBlur without having to pass in an onSubmit function from the parent. Is that possible?

**Comment by bochen2014** [comment]
@erikras Your first solution would bypass validation and submit the form, which cause the entire page to refresh as @goodbomb  mentioned.
the 2nd approach works perfectly.

I think to programmatically trigger a form to submit is something very likely to happen, e.g. i'm implementing a wizard, where the 'next' button is outside the form, hence I need to tirgger form submit from outside, and I couldn't find a way in docs. 

anyway, redux-form is awesome. thanks again for your great work!

**Comment by mehrdaad** [comment]
I'm having a problem calling ``const submitter = this.props.handleSubmit(this.props.onSubmit)
submitter() // submits`` causes exception: ``reduxForm.js?c2ae:121 Uncaught Error: You must either pass handleSubmit() an onSubmit function or pass onSubmit as a prop``

**Comment by bochen2014** [comment]
@mehrdaad  that's a very common error. the error is very self explainary:
You must either pass handleSubmit() an onSubmit function or pass onSubmit as a prop

option1: outside of your redux-form
```
<MyReduxFrom onSubmit={e=>{xxx}}
```

option2: inside your decorated form
```
render(){
  const {handleSubmit} =this.props
return(
  <form onSubmit={handleSubmit(e=>{xxx})}

</form>
}
```

**Comment by shotaK** [comment]
As I see there is a prop "triggerSubmit" coming, is it something we can use to submit forms programmatically?
